lisa/stats.py

```diff
@@ -840,6 +848,35 @@ def split_row_col(group):
             collapsed_col = None
             collapse_group = {}
 
+        if backend == 'plotly':
+            figure = plotly_make_subplots(
+                rows=nrows, cols=ncols,
+                subplot_titles=[
+                    '-'.join(group) if isinstance(group, tuple) else group
+                    for group, subdf in grouped
+                ]
+            )
+
+            for group, subdf in grouped:
+                group = fixup_tuple(group)
+                row, col = split_row_col(group)
+                group_dict = dict(zip(group_on, group))
+                x = rows.index(row) + 1
+                y = cols.index(col) + 1
+                subdf = subdf.drop(columns=group_on)
+                subdf = self._collapse_cols(subdf, collapse_group)
+                group_dict = dict(zip(group_on, group))
+                plot_func(subdf, (figure, x, y), collapsed_col,
```

Comment:
The value of having a `plot_func` callback lies in having a fixed type for the callback, i.e. a fixed signature, otherwise we end up with spaghetti. The "subplot" can become opaque (an axis for matplotlib or a tuple figure,x,y for plotly) but the rest should be the same.

While at it, you can make all parameters of plot_func keyword-only so there is no risk of mixup and we can easily add more in the future:
```
# foo and bar are keyword-only since the are after a lone start
def f(*, foo, bar): ...
```
https://www.python.org/dev/peps/pep-3102/

Once this is done, we may realize that the loop calling plot_func can actually be shared by both backends